src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -46,7 +49,15 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event)
             $subject = null;
 
             if ($subjectRef) {
-                if (\is_array($subjectRef)) {
+                if ($subjectRef instanceof Expression) {
+                    $this->expressionLanguage ??= class_exists(Expression::class)
+                        ? new ExpressionLanguage()
+                        : throw new \LogicException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed. Try running "composer require symfony/expression-language".')
+                    ;
+                    $subject = $this->expressionLanguage->evaluate($subjectRef, [
+                        'args' => $arguments,
```

Comment:
Using the ExpressionLanguage with the special functions registered in the security component but without passing the variables that those function expect is broken.

Reply:
Note that the authorization-related functions probably don't make sense in the subject expression (and they would suffer from the same issues than `@Security` in FrameworkExtraBundle by forcing to duplicate part of the logic of the authorization system, as those expression functions are meant to run in the ExpressionVoter inside the authorization system).